### PR TITLE
Updated the default region as the current one

### LINKED ISSUE #83: Use the current region by default
Currently, the default value of the region is "us-east1". The better solution is to use the current region of the EC2 Instance.

### PR DESCRIPTION
**Pull request by alparslanavci**
Fixes #83.

### CODE REVIEW

File: src/main/java/com/hazelcast/aws/AwsDiscoveryStrategy.java [hunk 1]
```diff
@@ -83,23 +85,45 @@ public AwsDiscoveryStrategy(Map<String, Comparable> properties) {
...
+
+    AwsConfig getAwsConfig()
             throws IllegalArgumentException {
+        Integer connectionRetries = getOrDefault(CONNECTION_RETRIES.getDefinition(), DEFAULT_CONNECTION_RETRIES);
+        Integer connectionTimeoutSeconds = getOrDefault(CONNECTION_TIMEOUT_SECONDS.getDefinition(),
+                DEFAULT_CONNECTION_TIMEOUT_SECONDS);
+        String region = getOrDefault(REGION.getDefinition(), null);
+        //to prevent unnecessary metadata call when region is set
+        if (region == null) {
+            region = getCurrentRegion(connectionTimeoutSeconds, connectionRetries);
```
Comment (leszko) [inline src/main/java/com/hazelcast/aws/AwsDiscoveryStrategy.java]: How about adding some logging here, like "Region not set, using the current region eu-central1". Or something similar?

File: src/main/java/com/hazelcast/aws/AwsDiscoveryStrategy.java [hunk 2]
```diff
@@ -83,23 +85,45 @@ public AwsDiscoveryStrategy(Map<String, Comparable> properties) {
...
+                                          .setConnectionRetries(connectionRetries)
                                           .setHzPort(new PortRange(getPortRange())).build();
 
         reviewConfiguration(config);
         return config;
     }
 
+    String getCurrentRegion(int connectionTimeoutSeconds, int connectionRetries) {
+        String uri = INSTANCE_METADATA_URI.concat(AVAILABILITY_ZONE_URI);
+        String availabilityZone = retrieveMetadataFromURI(uri, connectionTimeoutSeconds, connectionRetries);
```
[the remark this answers is not in the document]
Comment (alparslanavci) [reply src/main/java/com/hazelcast/aws/AwsDiscoveryStrategy.java]: Done 👍

File: src/main/java/com/hazelcast/aws/AwsDiscoveryStrategy.java (hunk 1 above)
Comment (alparslanavci) [reply src/main/java/com/hazelcast/aws/AwsDiscoveryStrategy.java]: Added 👍

**Comment by alparslanavci** [comment]
@leszko, I will check the other issues and plan a release.

**Comment by leszko** [comment]
Thanks a lot @alparslanavci

**Comment by mesutcelik** [comment]
I see `us-east-1` is mentioned in README and javadocs. Can we remove those? AFAIS, it is not relevant anymore. Do we still want to keep region [property](https://github.com/hazelcast/hazelcast-aws/blob/38f086d00b0987efb3726b414d1d9e12928e260c/src/main/java/com/hazelcast/aws/AwsProperties.java#L47)?

**Comment by mesutcelik** [comment]
I checked it again and I see default region is needed if hazelcast-client is running outside of AWS.

**Pull request merged**